core/src/main/java/tech/tablesaw/io/string/DataFramePrinter.java

```diff
@@ -189,22 +192,25 @@ private String tableName(Relation frame, int width) {
             int i;
             for (i = 0; i < Math.ceil((double) rowCount / 2); i++) {
                 for (int j = 0; j < colCount; j++) {
-                    data[i][j] = frame.getString(i, j);
+                    Column<?> col = frame.column(j);
+                    data[i][j] = (col.size() > i ? col.getString(i) : TOO_SHORT_COLUMN_MARKER);
                 }
             }
             for (int j = 0; j < colCount; j++) {
                 data[i][j] = "...";
             }
             for (++i; i <= rowCount; i++) {
                 for (int j = 0; j < colCount; j++) {
-                    data[i][j] = frame.getString(frame.rowCount() - maxRows + i - 1, j);
+                    Column<?> col = frame.column(j);
+                    data[i][j] = (col.size() > frame.rowCount() - maxRows + i - 1 ? col.getString(frame.rowCount() - maxRows + i - 1) : TOO_SHORT_COLUMN_MARKER);
```

Comment:
I'd probably refactor out `maxRows + i - 1` as a common variable. Though that will happen automatically if you make a common method as suggested below